cmd/web.go

```diff
@@ -51,6 +51,26 @@ and it takes care of all the other things for you`,
 	},
 }
 
+func runHTTPRedirector() {
+	source := fmt.Sprintf("%s:%s", setting.HTTPAddr, setting.PortToRedirect)
+	dest := strings.TrimSuffix(setting.AppURL, "/")
+	log.Info("Redirecting: %s to %s", source, dest)
+
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		target := dest + r.URL.Path
+		if len(r.URL.RawQuery) > 0 {
+			target += "?" + r.URL.RawQuery
+		}
+		http.Redirect(w, r, target, http.StatusTemporaryRedirect)
```

Comment:
Why Temporary and not Permanent? Then webbrowsers can go to https automagically afterwards even if they type http 🤔

Reply:
Temporary redirect also supports POST.  I based the redirect logic on this gist: https://gist.github.com/d-schmidt/587ceec34ce1334a5e60

I don't have strong feelings about this either way.  It felt more complete to support any type of request.  Chrome seems to treat temporary in a similar manner as permanent.  Presumably it is doing so with some kind of cache timeout, so you would go back to the http address after a period of time.

Reply:
Makes sense to me. LGTM